services/qs-spark-service/src/main/java/ai/quantumics/api/controller/AwsConnectionController.java

```diff
@@ -129,4 +132,21 @@ private ResponseEntity<Object> returnResInstance(HttpStatus code, String message
         genericResponse.put("message", message);
         return ResponseEntity.status(code).body(genericResponse);
     }
+
+    @PostMapping("/testConnection")
+    public ResponseEntity<Object> testConnection(@RequestBody @Valid AwsDatasourceRequest awsDatasourceRequest)
+            throws InvalidAccessTypeException {
+
+        dbUtil.changeSchema("public");
+        QsUserV2 user = validatorUtils.checkUser(awsDatasourceRequest.getUserId());
+        Projects project = validatorUtils.checkProject(awsDatasourceRequest.getProjectId());
+        dbUtil.changeSchema(project.getDbSchemaName());
+        List<Bucket> buckets = awsConnectionService.testConnection(awsDatasourceRequest);
+
+        if(CollectionUtils.isEmpty(buckets) || buckets.size() >= 0){
```

Comment:
test connection does not need check buckets size.. what if no buckets are created? is it valid to throw message